flink-connectors/flink-connector-pulsar/src/main/java/org/apache/flink/connector/pulsar/source/reader/PulsarPartitionReader.java

```diff
@@ -63,33 +61,33 @@ public PulsarPartitionSplit getSplit() {
         return split;
     }
 
-    public Iterator<Message<?>> nextBatch() throws PulsarClientException {
+    public Iterator<Message<V>> nextBatch() throws PulsarClientException {
         if (consumer.hasMessageAvailable()) {
-            Messages<byte[]> messages = consumer.batchReceive();
-            Iterator<Message<byte[]>> messageIterator = messages.iterator();
+            Messages<V> messages = consumer.batchReceive();
```

Comment:
Any reason for using `batchReceive()` here? The current implementation for the `batchReceive()` can’t improve performance for message reading. And the default behavior for the `BatchReceivePolicy` is to wait at most 100ms if there are not enough messages for a batch(default is 1MB).

Reply:
No reason for `batchReceive`, we would remove it.